ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -96,32 +104,84 @@ const onNavigationStateChange = (
 class IdpLoginScreen extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
-
     this.state = {
-      requestState: pot.noneLoading
+      requestState: pot.noneLoading,
+      startingMs: new Date().getTime()
     };
   }
 
-  private handleOnError = (): void =>
+  private getSpidId = (): string =>
+    this.props.loggedOutWithIdpAuth
+      ? this.props.loggedOutWithIdpAuth.idp.entityID
+      : "idpID n/a";
+
+  public componentDidMount() {
+    this.props.dispatch(
+      idpLoginStart({
+        id: this.getSpidId(),
+        detail: "start"
+      })
+    );
+  }
+
+  private handleOnError = (): void => {
+    this.props.dispatch(
+      idpLoginRequestError({
+        id: this.getSpidId(),
+        detail: "network request error"
+      })
+    );
     this.setState({
       requestState: pot.noneError("error")
     });
+  };
 
   private goBack = this.props.navigation.goBack;
 
-  private setRequestStateToLoading = () =>
+  private setRequestStateToLoading = (): void =>
     this.setState({ requestState: pot.noneLoading });
 
   private handleNavigationStateChange = (event: NavState): void => {
+    if (event.url && event.url !== this.state.requestUrl) {
+      this.props.dispatch(
+        idpLoginUrlChanged({
+          id: this.getSpidId(),
+          detail: event.url.split("?")[0]
+        })
+      );
+    }
+
     this.setState({
       requestState: event.loading
         ? pot.someLoading("loading")
-        : pot.some("loading complete")
+        : pot.some("loading complete"),
```

Comment:
if it's a string displayed to the user, needs i18n, or else use constant like identifier names